File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        longest = 0
        left = 0
        res = defaultdict(int)
        highest_freq = 0
        
        for right in range(len(s)):
            res[s[right]] += 1
            highest_freq = max(highest_freq, res[s[right]])
            
            window_size = right - left + 1
            if window_size - highest_freq > k:
                res[s[left]] -= 1
                left += 1
            
            longest = max(longest, right - left + 1)
        return longest

File: Data Structures & Algorithms/reverse-integer/submission-0.cpp
class Solution {
public:
    int reverse(int x) {
        int MAX=2147483647;
        int MIN=-2147483648;
        int res=0;
        while (x != 0) {
            if (res > MAX/10 || res < MIN/10) return 0;
            if (res == MAX/10 && x%10 > MAX%10) return 0;
            if (res == MIN/10 && x%10 < MIN%10) return 0;
            res = res*10 +(x%10);
            x /= 10;
        }
        return res;
    }
};
